arduino/CAN Forward/CAN_DEMO_FORWARD/CAN_DEMO_FORWARD.ino

```diff
@@ -0,0 +1,137 @@
+#include <driver/twai.h>
+#include <mcp_can.h>    // ADD library https://github.com/coryjfowler/MCP_CAN_lib
+#include <SPI.h>
+
+// CAN1 (TWAI) Pins
+#define CAN1_RX_PIN GPIO_NUM_6
+#define CAN1_TX_PIN GPIO_NUM_7
+
+// CAN2 (MCP2515) Custom SPI Pins
+#define CAN2_CS_PIN GPIO_NUM_10
+#define CAN2_SPI_SCK GPIO_NUM_12
+#define CAN2_SPI_MISO GPIO_NUM_13
+#define CAN2_SPI_MOSI GPIO_NUM_11
+
+SPIClass CAN2_SPI(HSPI); // HSPI bus
+MCP_CAN CAN2(&CAN2_SPI, CAN2_CS_PIN);
+
+void sendDataCAN2(byte *data, unsigned long data_len, unsigned long can_id)
+{
+
+  Serial.printf("Sending message to CAN2 [0x%03lX]: ", can_id);
+  for (byte i = 0; i < data_len; i++)
+  {
+    Serial.printf("0x%02X", data[i]);
+    if (i < data_len - 1)
+      Serial.print(", ");
+  }
+  Serial.println();
+
+  // Send the message via CAN2
+  byte result = CAN2.sendMsgBuf(can_id, 0, data_len, data);
+
+  if (result == CAN_OK)
+  {
+    Serial.println("Message sent successfully via CAN2");
+  }
+  else
+  {
+    Serial.print("Message send failed, code: ");
+    Serial.println(result);
+  }
+}
+
+void CAN1_readMsg()
+{
+  twai_message_t message;
+
+  while (twai_receive(&message, 0) == ESP_OK) // Non-blocking
+  {
+    // Print original CAN1 message
+    Serial.printf("[CAN1 0x%X]: ", message.identifier);
+    for (int i = 0; i < message.data_length_code; i++)
+    {
+      Serial.printf("0x%02X", message.data[i]);
+      if (i < message.data_length_code - 1)
+        Serial.print(", ");
+    }
+    Serial.println();
+
+    // Modify data (add 1 to each byte)
+    for (int i = 0; i < message.data_length_code; i++)
+    {
+      message.data[i] += 1;
+    }
+
+    // Send modified message to CAN2
+    sendDataCAN2(message.data, message.data_length_code, message.identifier);
+  }
+}
+
+bool setupCAN1(void)
+{
+  twai_general_config_t g_config = TWAI_GENERAL_CONFIG_DEFAULT((gpio_num_t)CAN1_TX_PIN, (gpio_num_t)CAN1_RX_PIN, TWAI_MODE_NORMAL);
+  twai_timing_config_t t_config = TWAI_TIMING_CONFIG_250KBITS();   // CAN1 at 500 kbps
+  twai_filter_config_t f_config = TWAI_FILTER_CONFIG_ACCEPT_ALL(); // Accept all messages
+
+  if (twai_driver_install(&g_config, &t_config, &f_config) != ESP_OK)
+  {
+    Serial.println("Failed to install TWAI driver");
+    return false;
+  }
+
+  if (twai_start() != ESP_OK)
+  {
+    Serial.println("Failed to start TWAI driver");
+    return false;
+  }
+
+  Serial.println("CAN1 (TWAI) started. Waiting for messages...");
+  return true;
+}
+
+bool setupCAN2()
+{
+  CAN2_SPI.begin(CAN2_SPI_SCK, CAN2_SPI_MISO, CAN2_SPI_MOSI, CAN2_CS_PIN);
+
+  if (CAN2.begin(MCP_STDEXT, CAN_500KBPS, MCP_16MHZ) == CAN_OK)
+  {
+    CAN2.setMode(MCP_NORMAL);
+    return true;
+  }
+  else
+  {
+    return false;
+  }
+}
+
+void setup()
+{
+  Serial.begin(115200);
+  while (!Serial)
+    ;
+  delay(5000);
+  // Setup CAN1 (TWAI) at 500kbps
```

Comment:
Change this to say "Setup CAN1 (TWAI)"

Remove the 500kbps - the function itself will define the baud rate.